cmd/translatesfx/translatesfx/otel.go

```diff
@@ -19,104 +19,175 @@ import (
 	"strings"
 )
 
-type otelCfg struct {
-	Extensions    map[string]interface{} `yaml:",omitempty"`
-	ConfigSources map[string]interface{} `yaml:"config_sources"`
-	Receivers     map[string]interface{}
-	Processors    map[string]interface{} `yaml:",omitempty"`
-	Exporters     map[string]interface{}
-	Service       map[string]interface{}
+const (
+	processlist       = "smartagent/processlist"
+	sfxFwder          = "smartagent/signalfx-forwarder"
+	resourceDetection = "resourcedetection"
+	sfx               = "signalfx"
+	metricsTransform  = "metricstransform"
+)
+
+func saInfoToOtelConfig(sa saCfgInfo) *otelCfg {
+	otel := newOtelCfg()
+	translateExporters(sa, otel)
+	translateMonitors(sa, otel)
+	translateGlobalDims(sa, otel)
+	translateSAExtension(sa, otel)
+	translateObservers(sa, otel)
+	return otel
 }
 
-const processlist = "smartagent/processlist"
+func newOtelCfg() *otelCfg {
+	return &otelCfg{
+		ConfigSources: map[string]interface{}{
+			"include": nil,
+		},
+		Receivers: map[string]map[string]interface{}{},
+		Processors: map[string]map[string]interface{}{
+			resourceDetection: {
+				"detectors": []string{"system", "env", "gce", "ecs", "ec2", "azure"},
+			},
+		},
+		Extensions: map[string]map[string]interface{}{},
+		Service: service{
+			Pipelines: map[string]*rpe{
+				"metrics": {
```

Comment:
Not required w/ refactoring but default agent config has forwarder as a metric receiver as well https://github.com/signalfx/splunk-otel-collector/blob/8c4639181c40e0454a07a41a30bd11315b2bede9/cmd/otelcol/config/collector/agent_config.yaml#L164

Reply:
I was thinking the forwarder could be added only if it was present in the SA config. Would it be better to always add it?